R/slice_derivation.R

```diff
@@ -83,20 +84,44 @@
 #'     args = params(time_imputation = "last")
 #'   )
 #' )
+#'
 slice_derivation <- function(dataset,
                              derivation,
                              args = NULL,
                              ...) {
   # check input
   assert_data_frame(dataset)
-  assert_function(derivation, params = c("dataset"))
+  assert_function(derivation)
   assert_s3_class(args, "params", optional = TRUE)
   if (!is.null(args)) {
     assert_function(derivation, names(args))
   }
   slices <- list2(...)
   assert_list_of(slices, "derivation_slice")
 
+  # Check that every mandatory argument to the derivation function is either passed
+  # inside args or present in every slice
+  mandatory_args <- formals(derivation) %>% keep(is_missing) %>% names()
+
+  if(length(mandatory_args > 1)){
+    mandatory_args <- mandatory_args[-1] # ignore first item, as that is the dataset
+
+    for (mandatory_arg in mandatory_args){
+
+      if (!mandatory_arg %in% names(args)){
+
+        check <- sapply(slices, function(x) mandatory_arg %in% x)
```

Comment:
Shouldn't it be
```suggestion
        check <- sapply(slices, function(x) mandatory_arg %in% names(x$args))
```
?